test/unit/ngsiv2/expressions/expressionBasedTransformations-test.js

```diff
@@ -0,0 +1,774 @@
+                lazy: [],
+                active: [
+
+                    {
+                        object_id: 'p',
+                        name: 'pressure',
+                        type: 'Integer',
+                        expression: '${trim(@spaces)}'
+                    },
+                    {
+                        object_id: 'p25',
+                        name: 'pressure25',
+                        type: 'Integer'
+                    },
+                    {
+                        object_id: 'e',
+                        name: 'consumption',
+                        type: 'Float',
+                        expression: '${trim(@spaces)}'
+                    },
+                    {
+                        object_id: 'h',
+                        name: 'humidity12',
+                        type: 'Percentage'
+                    },
+                    {
+                        name: 'weather',
+                        type: 'Summary',
+                        expression: 'Humidity ${@humidity12 / 2} and pressure ${@pressure25 * 20}'
+                    },
+                    {
+                        object_id: 'a',
+                        name: 'alive',
+                        type: 'Null',
+                        expression: '${trim(@spaces)}'
+                    },
+                    {
+                        object_id: 'u',
+                        name: 'updated',
+                        type: 'Boolean',
+                        expression: '${trim(@spaces)}'
+                    },
+                ]
+            }
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    };
+
+describe('Expression-based transformations plugin', function() {
+    beforeEach(function(done) {
+        logger.setLevel('FATAL');
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            iotAgentLib.clearAll(function() {
+                iotAgentLib.addUpdateMiddleware(iotAgentLib.dataPlugins.attributeAlias.update);
+                iotAgentLib.addQueryMiddleware(iotAgentLib.dataPlugins.attributeAlias.query);
+                iotAgentLib.addUpdateMiddleware(iotAgentLib.dataPlugins.expressionTransformation.update);
+                done();
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        iotAgentLib.clearAll(function() {
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When an update comes for expressions with syntax errors', function() {
+        // Case: Update for an attribute with bad expression
+        var values = [
+            {
+                name: 'p',
+                type: 'centigrades',
+                value: '52'
+            }
+        ];
+
+        beforeEach(function() {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
```

Comment:
Probably in this case you don't need the mock, as the fail this test is testing avoids any update to CB, isn't it?

Reply:
Fixed in 4026d75714dd6ef51230bd01c8482c0a89e36923